src/common/containers/repository-setup.js

```diff
@@ -5,39 +5,40 @@ import { withRouter } from 'react-router';
 
```

Comment:
Non-blocking comment:

I wonder whether this component needs to exist. If the repository information in the store is complete enough for a webhook and a build to be created, these actions should be initiated when the **createSnap** action successfully resolves. The spinner could also be moved to the RepositoryInput component.

Further, we should probably look at creating an API endpoint to aggregate the creation of webhooks and builds.